postgresql-persistence/src/main/java/com/netflix/conductor/aurora/AuroraExecutionDAO.java

```diff
@@ -0,0 +1,691 @@
+			.executeAndFetch(Task.class));
+	}
+
+	@Override
+	public List<Task> getPendingTasksForTaskType(String taskType) {
+		String SQL = "SELECT t.json_data FROM task_in_progress tip " +
+			"INNER JOIN task t ON t.task_id = tip.task_id " +
+			"WHERE tip.task_def_name = ?";
+
+		return queryWithTransaction(SQL,
+			q -> q.addParameter(taskType).executeAndFetch(Task.class));
+	}
+
+	@Override
+	public List<Task> getPendingSystemTasks(String taskType) {
+		String SQL = "SELECT t.json_data FROM task_in_progress tip " +
+			"INNER JOIN task t ON t.task_id = tip.task_id " +
+			"WHERE t.task_type = ?";
+
+		return queryWithTransaction(SQL,
+			q -> q.addParameter(taskType).executeAndFetch(Task.class));
+	}
+
+	@Override
+	public List<Task> getTasks(String taskType, String startKey, int count) {
+		List<Task> tasks = Lists.newLinkedList();
+
+		List<Task> pendingTasks = getPendingTasksForTaskType(taskType);
+		boolean startKeyFound = startKey == null;
+		int foundCount = 0;
+		for (Task pendingTask : pendingTasks) {
+			if (!startKeyFound) {
+				if (pendingTask.getTaskId().equals(startKey)) {
+					startKeyFound = true;
+					continue;
+				}
+			}
+			if (startKeyFound && foundCount < count) {
+				tasks.add(pendingTask);
+				foundCount++;
+			}
+		}
+		return tasks;
+	}
+
+	@Override
+	public List<Task> createTasks(List<Task> tasks) {
+		List<Task> created = Lists.newLinkedList();
+
+		withTransaction(connection -> {
+			for (Task task : tasks) {
+
+				Preconditions.checkNotNull(task, "task object cannot be null");
+				Preconditions.checkNotNull(task.getTaskId(), "Task id cannot be null");
+				Preconditions.checkNotNull(task.getWorkflowInstanceId(), "Workflow instance id cannot be null");
+				Preconditions.checkNotNull(task.getReferenceTaskName(), "Task reference name cannot be null");
+
+				task.setScheduledTime(System.currentTimeMillis());
+
+				boolean taskAdded = addScheduledTask(connection, task);
+				if (!taskAdded) {
+					String taskKey = task.getReferenceTaskName() + task.getRetryCount();
+					if (logger.isDebugEnabled())
+						logger.debug("Task already scheduled, skipping the run " + task.getTaskId() +
+							", ref=" + task.getReferenceTaskName() + ", key=" + taskKey);
+					continue;
+				}
+				addTaskInProgress(connection, task);
+				updateTask(connection, task);
+
+				created.add(task);
+			}
+		});
+
+		return created;
+	}
+
+	@Override
+	public void updateTask(Task task) {
+		withTransaction(connection -> updateTask(connection, task));
+	}
+
+	@Override
+	public boolean exceedsInProgressLimit(Task task) {
+		TaskDef taskDef = metadata.getTaskDef(task.getTaskDefName());
+		if (taskDef == null) {
+			return false;
+		}
+
+		int limit = taskDef.concurrencyLimit();
+		if (limit <= 0) {
+			return false;
+		}
+
+		long current = getInProgressTaskCount(task.getTaskDefName());
+		if (current >= limit) {
+			return true;
+		}
+
+		logger.info("Task execution count for {}: limit={}, current={}", task.getTaskDefName(), limit,
```

Comment:
Should we log the value of `current` here or call `getInProgressTaskCount` again? Couldn't the response of that method be different between the 2 invocations?